packages/next/client/page-loader.js

```diff
@@ -107,30 +128,31 @@ export default class PageLoader {
     })
   }
 
-  async loadScript (route) {
+  async loadRoute (route) {
     await this.promisedBuildId
 
     route = this.normalizeRoute(route)
     let scriptRoute = route === '/' ? '/index.js' : `${route}.js`
 
-    const script = document.createElement('script')
+    const url = `${this.assetPrefix}/_next/static/${encodeURIComponent(
+      this.buildId
+    )}/pages${scriptRoute}`
+    this.loadScript(url, route)
+  }
 
+  loadScript (url, route) {
+    const script = document.createElement('script')
     if (process.env.__NEXT_MODERN_BUILD && 'noModule' in script) {
       script.type = 'module'
-      scriptRoute = scriptRoute.replace(/\.js$/, '.module.js')
+      url = url.replace(/\.js$/, '.module.js')
```

Comment:
This needs to handle if it's already passed a modern version of a script now AFAIK. Maybe not, but worth checking.

Reply:
Yep--I didn't notice this was causing problems because of the silent fallback to reload/SSR